src/coreclr/debug/di/rsfunction.cpp

```diff
@@ -606,6 +610,92 @@ HRESULT CordbFunction::CreateNativeBreakpoint(ICorDebugFunctionBreakpoint **ppBr
     return hr;
 }
 
+//-----------------------------------------------------------------------------
+// CordbFunction::DisableOptimizations
+//  Public method for ICorDebugFunction5::DisableOptimizations.
+//   Triggers a new JIT so the next time the function is called, it will be unoptimized.
+//
+// Parameters
+//   
+//
+// Returns:
+//   S_OK on success.
+//-----------------------------------------------------------------------------
+HRESULT CordbFunction::DisableOptimizations()
+{
+    PUBLIC_API_ENTRY(this);
+    FAIL_IF_NEUTERED(this);
+    ATT_REQUIRE_STOPPED_MAY_FAIL(GetProcess());
+
+    HRESULT hr = S_OK;
+
+    CordbProcess * pProcess = GetProcess();
+    RSLockHolder lockHolder(pProcess->GetProcessLock());
+
+    DebuggerIPCEvent event;
+    CordbAppDomain * pAppDomain = GetAppDomain();
+    _ASSERTE (pAppDomain != NULL);
+
+    pProcess->InitIPCEvent(&event, DB_IPCE_DISABLE_OPTS, true, pAppDomain->GetADToken());
+    event.DisableOptData.funcMetadataToken = m_MDToken;
+    event.DisableOptData.pModule = m_pModule->GetRuntimeModule();
+
+    lockHolder.Release();
+    hr = pProcess->m_cordb->SendIPCEvent(pProcess, &event, sizeof(DebuggerIPCEvent));
+    lockHolder.Acquire();
+
+    _ASSERTE(event.type == DB_IPCE_DISABLE_OPTS_RESULT);
+
+    return event.hr;
+}
+
+//-----------------------------------------------------------------------------
+// CordbFunction::AreOptimizationsDisabled
+//  Public method for ICorDebugFunction5::AreOptimizationsDisabled.
+//   Indicates whether this method had optimizations disabled already.
+//
+// Parameters:
+//   BOOL *pOptimizationsDisabled
+//   
+//
+// Returns:
+//   S_OK on success.
+//-----------------------------------------------------------------------------
+HRESULT CordbFunction::AreOptimizationsDisabled(BOOL *pOptimizationsDisabled)
+{
+    PUBLIC_API_ENTRY(this);
+    FAIL_IF_NEUTERED(this);
+    ATT_REQUIRE_STOPPED_MAY_FAIL(GetProcess());
+
+    HRESULT hr = S_OK;
+
+    if (pOptimizationsDisabled == NULL)
+    {
+        return E_INVALIDARG;
+    }
+
+    CordbProcess * pProcess = GetProcess();
+    RSLockHolder lockHolder(pProcess->GetProcessLock());
+
+    DebuggerIPCEvent event;
+    CordbAppDomain * pAppDomain = GetAppDomain();
+    _ASSERTE (pAppDomain != NULL);
+
+    pProcess->InitIPCEvent(&event, DB_IPCE_IS_OPTS_DISABLED, true, pAppDomain->GetADToken());
```

Comment:
If the process is stopped to be able to do this inspection - is there any reason to not ask this question though the DAC?

Reply:
Since this is already tested and ready to go my plan is to check it in as is and do a follow up PR to address this and the GetFunctionAddress question